### PR TITLE
CAI-712: Added option to control response pretty formatting

### PR DESCRIPTION
**Pull request by mateuszgryc**
Why: 
As part of our profiling we found out that indeting part of response took ~30% of whole memory used by our service. 

Solution:
An additional option for new Routers to disable pretty formatting.

### CODE REVIEW

File: jsonrest.go [hunk 1]
```diff
@@ -160,10 +163,19 @@ func WithNotFoundHandler(h http.Handler) Option {
...
+	return func(r *Router) {
+		r.jsonPrettyFormat = false
+	}
+}
+
 // NewRouter returns a new initialized Router.
 func NewRouter(options ...Option) *Router {
 	hr := httprouter.New()
 	r := &Router{router: hr}
+	r.jsonPrettyFormat = true
```
Comment (thoeni) [inline jsonrest.go:178]: If you call this `jsonPrettyFormatDisabled` you can leverage the concept in Go of zero-values (boolean is false) which means that you don't need to specify anything - everything else stays pretty much the same, but `WithDisabledJSONPrettyFormat` will set it to true.

File: jsonrest.go [hunk 2]
```diff
@@ -307,7 +320,9 @@ func sendJSON(w http.ResponseWriter, status int, v interface{}) {
 	}
 
 	enc := json.NewEncoder(w)
-	enc.SetIndent("", "  ")
+	if r.jsonPrettyFormat {
+		enc.SetIndent("", "  ")
+	}
```
Comment (thoeni) [inline jsonrest.go:325]: And in that case this would become
```suggestion
	if !r.jsonPrettyFormatDisabled {
		enc.SetIndent("", "  ")
	}
```

**Review by thoeni — APPROVED** [review]
Looks good, just suggested a way you could use the zero value for the flag.

The important thing anyway - which I see you've done - is to make sure the library maintains the existing behaviour if the option is not specified, so that's good!

File: jsonrest.go (hunk 1 above)
Comment (mateuszgryc) [reply jsonrest.go:178]: Thanks for suggestion! To be honest that was my first approach for it in the end I thought that adding double negation `if !r.jsonPrettyFormatDisabled` is less readable than just default value true for enabling. Let me know if you don't agree with me and I can change it back

Comment (johngibb) [reply jsonrest.go:178]: Not a strong opinion, but `DisableJSONIndent` feels more in line with the stdlib's docs and doesn't read as poorly as a double negative, at least to me.

Comment (mateuszgryc) [reply jsonrest.go]: Thanks guys! It seems that it might actually make sense to rename it and enable zero value for it.

**Pull request merged**